sdk/resourcemanager/azure-resourcemanager-network/src/main/java/com/azure/resourcemanager/network/implementation/ApplicationGatewayImpl.java

```diff
@@ -814,6 +814,14 @@ public ApplicationGatewayImpl withTier(ApplicationGatewayTier skuTier) {
             this.innerModel().withSku(new ApplicationGatewaySku().withCapacity(1));
         }
         this.innerModel().sku().withTier(skuTier);
+        if (skuTier == ApplicationGatewayTier.WAF_V2 && this.innerModel().webApplicationFirewallConfiguration() == null) {
```

Comment:
If customer cannot set it, we will need a pending task to open it for user configure in convenience code. @XiaofeiCao

Reply:
Currently it can be se via https://github.com/Azure/azure-sdk-for-java/blob/df98d84ad0c7d90d23885d272d40b98a5393ff1b/sdk/resourcemanager/azure-resourcemanager-network/src/main/java/com/azure/resourcemanager/network/implementation/ApplicationGatewayImpl.java#L696-L706

Reply:
OK, @haolingdong-msft would you update the javadoc of the `withWebApplicationFirewall` to say that it is required for `WAF_V2`?

Reply:
Added.